### DISCUSSION TITLE
Takeover Mode (Vue)

### DISCUSSION THREAD

**Discussion by angelhdzmultimedia** [vuejs/language-tools#3670] [Q&A] [answered]
I'm a bit confused.

With Volar v2 (Vue), will we not have to disable built-in VSCode TypeScript extension anymore?

I had a hard time when making a new Vue project, and then remembering later to disable the built-in TS extension per-project, and it started to feel anti-DX. So I disabled Takeover mode, and just installed Volar TS extension instead, and the DX has been great even for Nuxt 3 (plus the convenient extension to restart TS lang server or Volar lang server).

But I read something that gave me the notion that this will not be needed in Volar v2.

Any info will be appreciated. 🤝🔥🩵💙

**Comment by so1ve** [comment, marked as answer]
Yes - volar used to require users to disable built-in typescript language server so that there will be only one language server for both vue and js/ts files. In volar 2.0, volar will not start a seperated language server for vue files, it will provide a "language plugin" to reuse the built-in typescript language server.

**Reply by angelhdzmultimedia** [reply]
Thank you for your time, information provided, and kindness.

**Reply by Random-nobody** [reply]
Would be good to add some information like this to the readme for the extension...

Right the details page for the extension is saying literally nothing about what is it for, how to use it, if there is configuration needed. So you find out about stuff like take-over mode the hard way. And unless you find external tutorial leading you to this extension, you wouldn't know what is it for. This is really confusing.
![2024-06-01_20-37-37_Code](https://github.com/vuejs/language-tools/assets/16503469/61a4ac41-eb1c-4b09-9466-cfc57c0d6109)
